src/fetchTileData.ts

```diff
@@ -0,0 +1,60 @@
+// Adapted from https://github.com/perliedman/node-hgt/blob/master/src/hgt.js
+
+import fnv1a from '@sindresorhus/fnv1a';
+import {LRUCache} from 'lru-cache';
+
+// Generic type for a function that somehow fetches a tile and produces a `Buffer`, e.g.,
+// `s3Fetcher` fetches a tile from an S3 bucket.
+type Fetcher<T extends unknown[]> = (...args: T) => Promise<Buffer>;
+
+// Thrown by Fetchers to indicate a missing tile, which may (e.g. elevation) or may not be expected.
+export const TILE_MISSING = Symbol();
+
+// Generic error thrown by Fetchers to that signal something went wrong.
+export const BAD_TILE = Symbol();
+
+// ...could signal other Fetcher error conditions, e.g. bad permissions, network failure, etc.
+
+
+const cache = new LRUCache<bigint, Buffer>({
+    sizeCalculation: (buffer) => buffer.length, // bytes
+    maxSize: 500_000_000, // 500 MB, specified in bytes because `sizeCalculation` returns bytes
```

Comment:
should this be an env var to be able to easily tune?

Reply:
That's definitely a good call. Will implement.